src/drt/src/io/GuideProcessor.cpp

```diff
@@ -369,80 +369,270 @@ void connectGuidesWithBestPinLoc(const Point3D& guide_pt,
   }
 }
 
-}  // namespace
+/**
+ * @brief logs the number of guides read so far
+ *
+ * If the number of guides read is less than 1M guides, the function reports
+ * every 100k guides, otherwise it reports every 1M guides read
+ * @param num_guides number of guides read so far
+ */
+void logGuidesRead(const int num_guides, utl::Logger* logger)
+{
+  bool log = false;
+  if (num_guides < 1000000) {
+    log = (num_guides % 100000 == 0);
+  } else {
+    log = (num_guides % 1000000 == 0);
+  }
+  if (log) {
+    logger->info(DRT, 156, "guideIn read {} guides.", num_guides);
+  }
+}
+/**
+ * @brief Checks the validity of the odb guide layer
+ *
+ * The db_guide layer is invalid if it is any of the following conditions:
+ * - Not in the DRT layer database
+ * - Above the sepecified top routing layer
+ * - Below the specified bottom routing layer and the via access layer
+ * @note If a layer is invalid, this produces an error unless it is above the
+ * top routing layer for a net that has pins above the top routing layer. In the
+ * latest case, we just ignore the guide and the pin is handled by
+ * io::Parser::setBTerms_addPinFig_helper
+ * @param layer_num The layer_num of the guide returned by this function if it
+ * is a valid layer
+ * @returns True if the guide is valid by the previous criteria and False
+ * otherwise
+ */
+bool isValidGuideLayerNum(odb::dbGuide* db_guide,
+                          frTechObject* tech,
+                          frNet* net,
+                          utl::Logger* logger,
+                          frLayerNum& layer_num)
+{
+  frLayer* layer = tech->getLayer(db_guide->getLayer()->getName());
+  if (layer == nullptr) {
+    logger->error(
+        DRT, 154, "Cannot find layer {}.", db_guide->getLayer()->getName());
+    return false;
```

Comment:
There is no need to return after error() which will throw an exception rendering it unreachable.